pkg/option/config.go

```diff
@@ -2690,6 +2690,10 @@ func (c *DaemonConfig) Populate() {
 		}
 	}
 
+	if nativeRoutingCIDR == "" && ipv4NativeRoutingCIDR == "" && c.EnableAutoDirectRouting {
+		log.Warnf("%s will be ineffective since %s is not set", EnableAutoDirectRoutingName, IPv4NativeRoutingCIDR)
```

Comment:
I don't think that's actually what we're saying. `--auto-direct-node-routes` will still be effective, that logic only creates routes on the nodes based on k8s pod CIDR ranges for each node.

The concern is that traffic between pods on different nodes could be unnecessarily NAT'd, which can cause problems with performance, observability and policy.